### ISSUE TITLE
Tests simultaneously pass and fail in benchdnn perf mode

### ISSUE THREAD

**Issue by jondea** [oneapi-src/oneDNN#1302]
# Summary
Tests simultaneously pass and fail in `benchdnn` perf mode, for example
```
tests/benchdnn/benchdnn --binary --mode=p 1:1
```
returns
```
0:EXECUTED (errors:0 total:0) __REPRO: --binary --mode=P 1:1
Output template: perf,%engine%,%impl%,%name%,%prb%,%Gops%,%-time%,%-Gflops%,%0time%,%0Gflops%
perf,cpu,ref:any,,--binary --mode=P 1:1,<data removed>
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 failed:1 listed:0
total perf: min(ms):<data removed> avg(ms):<data removed>
```
along with a non-zero exit code.

I think it is a bug that the tests both pass and fail, although I wasn't sure what the correct status should be.

I think it's caused by a missing break statement at https://github.com/oneapi-src/oneDNN/blob/05c51418b67f555c5dd6032a5c6210c70735c44c/tests/benchdnn/common.cpp#L102-L110 introduced in 05c51418b67f555c5dd6032a5c6210c70735c44c.

# Version
oneDNN v2.6.0 (commit df1788867266f4bcbe91ff2d10404759e1ef83d6) current master, although it applies from 05c51418b67f555c5dd6032a5c6210c70735c44c

# Environment
Any

**Comment by dzarukin** [comment]
Hi @jondea, thanks for reporting the issue. It's clearly a bug. Will fix it soon.

**Comment by dzarukin** [comment]
Hi @jondea, the issue should be fixed by [421606eb](https://github.com/oneapi-src/oneDNN/commit/421606eb196b6b9bd47836b7262620f12a05d769). Thank you.

**Comment by jondea** [comment]
@dzarukin just tested on 421606eb196b6b9bd47836b7262620f12a05d769 and it now passes, thanks!
